daemon/response.go

```diff
@@ -270,6 +270,115 @@ func (f fileResponse) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	http.ServeFile(w, r, string(f))
 }
 
+type followChangeJSON struct {
+	Id      string `json:"id"`
+	Kind    string `json:"kind"`
+	Summary string `json:"summary"`
+	Ready   bool   `json:"ready"`
+	Status  string `json:"status"`
+	// XXX: do we want/need this?
+	OldStatus string `json:"old-status"`
+	// XXX: spawn/ready time?
+}
+
+type followChangesSeqResponse struct {
+	activeChanges map[*state.Change]state.Status
+
+	chgsCh chan *followChangeJSON
+}
+
+func newFollowChangesSeqResponse(st *state.State) *followChangesSeqResponse {
+	rsp := &followChangesSeqResponse{
+		activeChanges: make(map[*state.Change]state.Status),
+		// XXX: use a small buffer size
+		chgsCh: make(chan *followChangeJSON, 10),
+	}
+	st.AddTaskStatusChangedObserver(rsp.onTaskChanged)
+
+	return rsp
+}
+
+// XXX: better name?
+func (fc *followChangesSeqResponse) Stop() {
+	close(fc.chgsCh)
```

Comment:
This is not synchronized with the send in `onTaskChanged`. Unless you need to stop this it's fine to let the channel garbage collect.